dbt_copilot_helper/templates/addons/env/rds-postgres.yml

```diff
@@ -211,28 +213,57 @@ Resources:
       AllowMajorVersionUpgrade: false
       AutoMinorVersionUpgrade: true
       BackupRetentionPeriod: 8
-      EnablePerformanceInsights: true
+      EnablePerformanceInsights:
+        !If [
+          {{ addon_config.prefix }}UseSnapshot,
+          !Ref AWS::NoValue,
+          true
+        ]
```

Comment:
Why has this become conditional?

Reply:
And the stuff further down, why are we using `AWS::NoValue` if we are using snapshots?

Reply:
Certain fields (MasterPassword, MasterUsername and the others in here) are mutually exclusive with DBSnapshotId. However the definitions are global to all envs, so cannot be removed for just one env.

Hence conditionals. AWS::NoValue makes CloudFormation treat the field as if it isn't there (unlike setting it to an empty string for example)

(It was the same for the Aurora DR if you take a look, except for slightly different fields due to it being a DB rather than a cluster)